### PR TITLE
secrets-store: add test jobs to validate migration to community cluster

### PR DESCRIPTION
**Pull request by aramase**
Adding test jobs `pull-secrets-store-csi-driver-e2e-windows-prow-migration-test` and `pull-secrets-store-csi-driver-e2e-azure-prow-migration-test` to validate move to community prow cluster + change in azure tenant.

### CODE REVIEW

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml [hunk 1]
```diff
@@ -658,6 +658,92 @@ presubmits:
...
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20240705-131cd74733-master
+        command:
+          - runner.sh
+        args:
+          - bash
+          - -c
+          - >-
+            ./test/scripts/run-e2e-windows.sh
```
Comment (aramase) [inline config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:732]: As part of the move to the new tenant for azure tests, we don't want to create a SP with secret. My current plan is to create an AKS cluster, MI and configure FIC. I plan to do all that in 1 script that'll handle azure linux and windows tests, so using the existing script now.

Comment (nilekhc) [reply config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:732]: @aramase Why are we not replacing existing job `pull-secrets-store-csi-driver-e2e-windows`?

Comment (aramase) [reply config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:732]: I want the CI to be functional until we complete the testing for migration. This also requires changes in driver, so once we finish the validation, we can replace existing jobs with the changes.

**Pull request merged**